### PR TITLE
refactor: make use of StatusQ `StatusAppLayout`

### LINKED ISSUE #2688: Use StatusQ Layout components to lay out app views
The second step in the migration process to moving to StatusQ components, is to use the layout framework provided by StatusQ. 

This entails a bunch of things:

1. Make use of `StatusAppLayout` which holds a `StatusAppNavBar`
1. Ensure community tab buttons are properly dynamically added
1. Also ensure app section switch continues to work
1. App layout also expects `StatusAppTwoPanelLayout` components for its views, so all "app sections" (Chat, community, profile, wallet etc) should use that component on their most outer level. This means we have to touch all `*Layout` components that live in Status Desktop at the moment

Last but not least: there's a bunch of app section specific functionality and components attached to `AppLayout` that are expected to exist throughout the entire application (another thing that needs a refactor, but we leave that for another step). 

Those specific functionalities need to continue to work, so let's make sure we don't break anything.

### PR DESCRIPTION
**Pull request by PascalPrecht**
This first step in leveraging StatusQ components. This sets up the overall
app layout, allowing us to use other sub view layouts provided by StatusQ,
such as `StatusAppTwoPanelLayout`.

This commit primarily touches the application navbar, leaving layout changes
in dedicated views for future commits.

Partially closes #2688

### CODE REVIEW

File: ui/app/AppMain.qml [hunk 1]
```diff
@@ -100,7 +105,209 @@ RowLayout {
...
+                }
+
+                StatusMenuItem {
+                    text: qsTr("View Community")
+                    icon.name: "group"
+                    onTriggered: openPopup(communityMembersPopup, {community: chatsModel.communities.observedCommunity})
+                }
+
+                StatusMenuItem {
+                    enabled: false//chatsModel.communities.observedCommunity.admin
```
Comment (jrainville) [inline ui/app/AppMain.qml:167]: Leftover?

File: ui/app/AppMain.qml [hunk 2]
```diff
@@ -100,7 +105,209 @@ RowLayout {
...
+                    onTriggered: chatsModel.communities.leaveCommunity(model.id)
+                }
+            }
+        }
+
+        navBarTabButtons: [
+            StatusNavBarTabButton {
+                icon.name: "wallet"
+                tooltip.text: qsTr("Wallet")
+                /* enabled: isExperimental === "1" || appSettings.isWalletEnabled */
```
Comment (jrainville) [inline ui/app/AppMain.qml:190]: Need to add back I think

**Review by jrainville — COMMENTED** [review]
Nice work. I think you have 2 leftovers

File: ui/app/AppMain.qml (hunk 1 above)
Comment (PascalPrecht) [reply ui/app/AppMain.qml:167]: Nope, I've just kept the comment in place. See: https://github.com/status-im/status-desktop/pull/2699/files#diff-0067538414dc279244337c270963fba7e96ad02472aa3c18d1007f3ceaa8ce81L57

File: ui/app/AppMain.qml (hunk 2 above)
Comment (PascalPrecht) [reply ui/app/AppMain.qml:190]: Yes, good catch!

**Review by jrainville — APPROVED** [review]

**Pull request merged**